accounts-db/src/append_vec.rs

```diff
@@ -348,9 +376,26 @@ impl AppendVec {
 
     /// when we can use file i/o as opposed to mmap, this is the trigger to tell us
     /// that no more appending will occur and we can close the initial mmap.
+    #[allow(dead_code)]
     pub(crate) fn reopen_as_readonly(&self) -> Option<Self> {
-        // this is a no-op when we are already a mmap
-        None
+        #[cfg(not(unix))]
+        // must open as mmmap on non-unix
+        return None;
+
+        #[cfg(unix)]
+        match &self.backing {
+            // already a file, so already read-only
+            AppendVecFileBacking::FileOnly(_file) => None,
+            AppendVecFileBacking::MmapOnly(_mmap) => {
+                // we are a map, so re-open as a file
+                _ = self.flush();
```

Comment:
I don't know what to do. Should we panic? Try again?

Reply:
What is the goal of flush here? If the goal is for new_from_file to see the
changes, it can already see them without flushing

Reply:
> What is the goal of flush here? If the goal is for new_from_file to see the
> changes, it can already see them without flushing

are you sure? I am tracking a bug in 1.18 right now that smells like a flush could have not been successful somehow.